### PR TITLE
[JENKINS-56393][JENKINS-56394] RfLintParser improvements

### PR DESCRIPTION
**Pull request by bkhouri**
# Change Description
The RfLintParser was not displaying the correct category.  It was using the severity as the category, instead of the rule name.

This change ensure that when a violation/issue is reported, the rule name is used as the category

# Testing Done
1. Locally build the analysis-model project using `man clean install`
2. Replaced `${JENKINS_HOME}/plugins/analysis-model-api/WEB-INF/lib/analysis-model-x.y.x.jar` with `<repo>/target/analysis-model-<version>.jar`
3. Trigger a job and ensure the categories are updated correctly

# Before
Before

# After
After - Category
After - Types

### CODE REVIEW

**Comment by uhafner** [comment]
Isn't `category` also the wrong property? Shouldn't the rule be the `type` (as this is done for the other parsers as well)? Category is something like 'Spelling', 'Documentation', etc. 

Example PMD: https://github.com/pmd/pmd/wiki/Rule-Categories

**Comment by bkhouri** [comment]
Update the code so that type is either `Suite`, `Test Case`, `Keyword`, `Other` or `Custom` and set the rule name as the type.

**Comment by bkhouri** [comment]
Is there a way we can retrigger the builds?

File: src/main/java/edu/hm/hafner/analysis/parser/RfLintParser.java [hunk 1]
```diff
@@ -57,30 +129,34 @@ public Report parse(final ReaderFactory readerFactory) {
...
+        String message = matcher.group("message");
+        String severityStr = guessCategoryIfEmpty(matcher.group("severity"), message);
+        String ruleName = matcher.group("ruleName");
+        RfLintCategory category = RfLintCategory.CUSTOM;
+        try {
+            RfLintRuleName rule = RfLintRuleName.valueOf(ruleName);
+            category = rule.getCategory();
+        } catch (IllegalArgumentException e) {
+            // do nothing
+        }
```
Comment (uhafner) [inline src/main/java/edu/hm/hafner/analysis/parser/RfLintParser.java]: Never use exceptions for control flow, better use `EnumUtils.isValidEnum(MyEnum.class, myValue)`. Moreover, that code belongs to the enum (tell don't ask).

File: src/main/java/edu/hm/hafner/analysis/parser/RfLintParser.java [hunk 2]
```diff
@@ -57,30 +129,34 @@ public Report parse(final ReaderFactory readerFactory) {
...
-                priority = Severity.WARNING_LOW;
-                category = "IGNORE";
-                break;
-            default:
-                break;
+        try {
+            RfLintSeverity rfLintSeverity = RfLintSeverity.valueOf(severityStr.substring(0, 1));
+            priority = rfLintSeverity.getSeverityLevel();
+        } catch (IllegalArgumentException e) {
+            // do nothing
```
Comment (uhafner) [inline src/main/java/edu/hm/hafner/analysis/parser/RfLintParser.java]: Same as above, move to enum.

**Comment by uhafner** [comment]
Can you please also add a Changelog entry?

File: src/main/java/edu/hm/hafner/analysis/parser/RfLintParser.java (hunk 1 above)
Comment (bkhouri) [reply src/main/java/edu/hm/hafner/analysis/parser/RfLintParser.java]: Updated to use `EmailUtils.isValidEnum`

File: src/main/java/edu/hm/hafner/analysis/parser/RfLintParser.java (hunk 2 above)
Comment (bkhouri) [reply src/main/java/edu/hm/hafner/analysis/parser/RfLintParser.java]: Updated to use `EmailUtils.isValidEnum`

**Comment by bkhouri** [comment]
Changlog entry added.

File: src/main/java/edu/hm/hafner/analysis/parser/RfLintParser.java (hunk 1 above)
Comment (uhafner) [reply src/main/java/edu/hm/hafner/analysis/parser/RfLintParser.java]: It still should be more object oriented a part of the Enum, e.g. as method 
```
RfLintCategory.fromCategory(final String category)
```
or something similar.

**Comment by uhafner** [comment]
Can you please check https://app.codacy.com/app/uhafner/analysis-model/pullRequest?prid=3238767. Then we are done :-)

**Comment by bkhouri** [comment]
Addressed, but I don't the codacy build being retriggered, at least GitHub isn't showing the link.

**Comment by uhafner** [comment]
Thank you very much!

**Comment by bkhouri** [comment]
I was happy to contribute, and thank you very much for the feedback you provided.  I started off with code, and ended up with much better and more robust code!

**Pull request merged**